### DISCUSSION TITLE
class from property not showing

### DISCUSSION THREAD

**Discussion by IPreferWater** [tailwindlabs/tailwindcss#11302] [Help] [answered]
Hello,

I have a simple component that received some **props**
one of the props is the **color** that will be used 

```
  export function TimetableEvent({event}:EventProps) {

    return (
      <div className={`bg-${event.color}`}>
       [...]
      </div>
)
```

1] The data are stored in a json with values like **"color": "blue-300"** or **"color": "red-300"**

it's not showing in the html, altought we can see the class bg-blue-300 in the div

![tailwind_1](https://github.com/tailwindlabs/tailwindcss/assets/51187329/715b1184-8c71-4d79-946b-f3465fdf575c)

in this example the event "jujitsu" of monday and thuesday should be in blue-300
and the event "muay-thai" should be red-300

2] To check, I used **hard-coded** value

```
  export function TimetableEvent({event}:EventProps) {

    return (
      <div className={`bg-blue-300`}>
       [...]
      </div>
)
```

![tailwind_2](https://github.com/tailwindlabs/tailwindcss/assets/51187329/1d2ab02f-088e-4463-93d8-32d84e155712)

all the div have the class bg-blue-300, it's **working well**

3] And now this is where I can't figure whats happening, I **ctrl-z** my modifications and go back to the first code I used

```
<div className={`bg-${event.color}`}>
```

the div with bg-blue-200 are working now, but not the others ...

![tailwind_3](https://github.com/tailwindlabs/tailwindcss/assets/51187329/f8fc4e62-f99c-42a0-b54f-6feee93d3e61)

If I stop my nextjs application and reboot, they will go back to the first state, not working

### is there something important in Tailwind I didn't understand ?

the [code's component](https://github.com/IPreferWater/ipreferwater-website/blob/feat/calendar/components/timetable/TimetableEvent.tsx#L10)

I already did something similar but instead of **building** the bg-color I used a switch that return a **const value**
[the code](https://github.com/IPreferWater/cel-nextjs/blob/master/components/plannings/PlanningDay.tsx#L11) and the [webpage ](https://www.clementinestla.com/plannings/2023_6)in action

Thank you in advance, I really enjoy working with Tailwindcss

**Comment by wongjn** [comment, marked as answer]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
>
> **Don’t construct class names dynamically**
>
> ```vue
> <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> ```
> In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> Instead, make sure any class names you’re using exist in full:
>
> **Always use complete class names**
>
> ```vue
> <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> ```

**Comment by IPreferWater** [comment]
thank you I totally missed this important part
